api/storage_proxy.cc

```diff
@@ -201,29 +202,38 @@ void set_storage_proxy(http_context& ctx, routes& r) {
     });
 
     sp::get_hinted_handoff_enabled.set(r, [&ctx](std::unique_ptr<request> req)  {
-        auto enabled = ctx.db.local().get_config().hinted_handoff_enabled();
-        return make_ready_future<json::json_return_type>(enabled.to_configuration_string());
+        auto filter = service::get_storage_proxy().local().get_hints_host_filter();
+        return make_ready_future<json::json_return_type>(!filter.is_disabled_for_all());
     });
 
     sp::set_hinted_handoff_enabled.set(r, [](std::unique_ptr<request> req)  {
-        //TBD
-        unimplemented();
         auto enable = req->get_query_param("enable");
-        return make_ready_future<json::json_return_type>(json_void());
+        auto filter = (enable == "true")
+                ? db::hints::host_filter(db::hints::host_filter::enabled_for_all_tag {})
+                : db::hints::host_filter(db::hints::host_filter::disabled_for_all_tag {});
+        return service::get_storage_proxy().invoke_on_all([filter] (service::storage_proxy& sp) {
```

Comment:
Let's here at least separate the enable/disable flag from the list.

Perhaps we should also implement the configuration with a separate item (but keep compatibility with ourselves).

Reply:
I'm not sure if I understand this comment, and if it still applies...

There are two API endpoints for settings hints configuration at the moment: `set_hinted_handoff_enabled` and `set_hinted_handoff_enabled_by_dc_list`. First takes a "true/false" flag and enables/disables hints completely, and the second one enables hints selectively for a given list of DCs. Therefore there is some kind of separation on the HTTP API level.

The YAML configuration does mix up enabling/disabling hints altogether in a single option, but that's because we inherited it from Cassandra (see my comment [here](https://github.com/scylladb/scylla/pull/6916#discussion_r468422079)). We decided to change it another time.